### ISSUE TITLE
Switch Vagrant hosts updating plugin?

### ISSUE THREAD

**Issue by jeremyfelt** [Varying-Vagrant-Vagrants/VVV#606]
[vagrant-hostsupdater](https://github.com/cogitatio/vagrant-hostsupdater) hasn't been updated in a while and may or may not be stable with current environments. See #603 as an example.

10up has a fork of vagrant-hostsupdater in [Vagrant Ghost](https://github.com/10up/vagrant-ghost). @ericmann any thoughts on that being a stable replacement?

@zamoose Does HGV have any plans along these lines? I did a search and only ran into one related issue ([HGV#40](https://github.com/wpengine/hgv/issues/40)).

**Comment by ericmann** [comment]
I (obviously) have no objections here. For context, vagrant-hostsupdater was _itself_ a fork of yet another unmaintained project (that broke when Vagrant changed the plugins API version a while back). We forked yet again so we could polish off some unimplemented features of the plugin.

Namely, command line usage.

In addition to setting the `ghost.config.hosts` array in a Vagrantfile, you can also update an `aliases` file in the project, then run `vagrant ghost` from the command line to automatically ingest those new host aliases into the machine (and your local hosts file). It's a simple change, but it was a command that worked in the original hostsmanager plugin, was abandoned in hostsupdater, and was required for our use.

I can definitely put together a PR for this change.

**Note:** @johnpbloch did the original forking work, but the plugin had to be co-located in another repo for testing. He lost props for his efforts when I migrated things into its own repository. I point this out to emphasize the project is supported by more than just me.

**Comment by EHLOVader** [comment]
This seems interesting. How far from the original hostsupdater does the 10up fork get? and would it still be possible to add the commands to the host's sudoers file to skip the password each time?

I believe I have a script that works with the plugins now. https://gist.github.com/EHLOVader/606f7c9f8717608527b8

**Comment by ericmann** [comment]
It cleans up a bit of the documentation, removes some dead code, and adds the ability to run `vagrant ghost` from the command-line to re-ingest your hosts and update the hosts file. 

The "original" hostsupdater plugin was itself a fork of another plugin. The real original plugin wasn't updated to the v2 of the Vagrant plugin API, so it stopped working a while ago. Hostsupdater forked and updated some of the code, but skipped the CLI entirely (even though it reports as being available with `vagrant list-commands`).

The entire point of the Ghost fork is to keep the plugin alive (there's no guarantee that Vagrant will stick with the v2 plugins API) and to rebuild the CLI components of the plugin. Aside from that, we also switched to referring to an `aliases` file instead of a `hosts` file for a list of addresses. This is because we often use Ansible for provisioning, and a `hosts` file has a completely different meaning in that world (the recursive directory search of VVV flat out breaks things if used in an Ansible project).

**Comment by johnpbloch** [comment]
On top of that, we also cleaned up how it behaves. The old plugin sometimes wouldn't add/remove hosts based on a variety of factors including what state the machine was transitioning into, or configuration options that prevented it from running on suspend by default. The new behavior is:

1. If the machine is transitioning into a "not reachable" state (i.e. `halt`, `suspend`, `destroy`), remove the hosts.
1. If it's transitioning into a "reachable" state (i.e. `up`, `resume`), add the hosts.
1. If it's doing a reload, wait for boot then remove the hosts and re-add them.

Basically, if the vm isn't up and running, the hosts should not be in your host file.

**Comment by cfoellmann** [comment]
> the recursive directory search of VVV flat out breaks things if used in an Ansible project
@ericmann does this mean it will break auto setup sites?

**Comment by cgsmith** [comment]
Hey @jeremyfelt , I'll be releasing some fixes to [vagrant-hostsupdater](https://github.com/cogitatio/vagrant-hostsupdater) as I was added to the repo.  I would love to include any contributions to help improve the project.  I will also look at pulling in [vagrant ghost](https://github.com/10up/vagrant-ghost) changes into the repo.

**Comment by jeremyfelt** [comment]
I'm going to close this out. [vagrant-hostsupdater](https://github.com/cogitatio/vagrant-hostsupdater/) is an active project at this point. Thanks for picking it up, @cgsmith! :)
